store/src/executor/storage_handlers.rs

```diff
@@ -54,10 +57,32 @@ impl RequestHandler<TruncateTableAction> for ActionHandler {
             .get(tbl_name)
             .ok_or_else(|| ErrorCode::UnknownTable(format!("table not found: {:}", tbl_name)))?;
 
-        self.meta_node
-            .remove_table_data_parts(db_name, tbl_name)
-            .await;
+        let cr = LogEntry {
+            txid: None,
+            cmd: Cmd::TruncateTable {
+                db_name: db_name.clone(),
+                table_name: tbl_name.clone(),
+            },
+        };
 
-        Ok(TruncateTableResult {})
+        let rst = self
+            .meta_node
+            .write(cr)
+            .await
+            .map_err(|e| ErrorCode::MetaNodeInternalError(e.to_string()))?;
+
+        match rst {
+            AppliedState::Table { prev, .. } => {
+                if prev.is_some() {
+                    Ok(TruncateTableResult {})
+                } else {
+                    Err(ErrorCode::UnknownTable(format!(
+                        "table not found: {:}",
+                        tbl_name
+                    )))
+                }
+            }
+            _ => Err(ErrorCode::MetaNodeInternalError("not a Table result")),
+        }
```

Comment:
For a succeeded apply, the `prev` and the `result` should be different to indicate a change.

Reply:
ok